searchkit/constraints.py

```diff
@@ -109,578 +107,691 @@ def __repr__(self):
-                         "- aborting constraints for this file.")
-                state.rc = state.RC_ERROR
-                return state
-
-            # until we get out of a skip range we want to leave the pos at the
-            # start but we rely on the caller to enforce this so that we don't
-            # have to seek(0) after every skip.
-            state.skip_current_line()
-            return state
-
-        if newpos is None:
-            state.rc = state.RC_FOUND_BAD
-            if state.cur_ln == 0:
-                log.debug("first line is not valid, checking last line")
-                state.cur_ln = state.search_range_end
-                state.update_pos_pointers()
-            elif (state.search_range_end - state.search_range_start) >= 1:
-                # _start_ln = state.search_range_start
-                state.search_range_start = state.cur_ln
-                # log.debug("going forwards (%s->%s:%s)", _start_ln,
-                #           state.search_range_start, state.search_range_end)
-                state.invalid_range.save_and_reset()
-                state.get_next_midpoint()
-                state.update_pos_pointers()
-        else:
-            state.save_last_known_good()
-            state.rc = state.RC_FOUND_GOOD
-            if state.cur_ln == 0:
-                log.debug("first line is valid so assuming same for rest of "
-                          "file")
-                self.fd_info.reset()
-            elif state.search_range_end - state.search_range_start <= 1:
-                log.debug("found final good ln=%s", state.cur_ln)
-                self.fd_info.fd.seek(state.cur_pos)
-            elif (len(state.invalid_range) > 0 and
-                  state.invalid_range.direction == SkipRange.SKIP_FWD):
-                log.debug("found good after skip range")
-                self.fd_info.fd.seek(state.cur_pos)
-            else:
-                # set rc to bad since we are going to a new range
-                state.rc = state.RC_FOUND_BAD
-                # _end_ln = state.search_range_end
-                state.search_range_end = state.cur_ln
-                # log.debug("going backwards (%s:%s->%s)",
-                #           state.search_range_start, _end_ln,
-                #           state.search_range_end)
-                self.fd_info.fd.seek(state.cur_pos)
-                state.invalid_range.save_and_reset()
-                state.get_next_midpoint()
-                state.update_pos_pointers()
-
-        return state
-
-    def _seek_to_first_valid(self, destructive=True):
+        @return:
+            FindDelimiterResult(FindTokenStatus.FOUND, ...) if token is found
+            FindDelimiterResult(FindTokenStatus.REACHED_EOF, ...) if token is
+                not found because the scan reached the EOF
+            FindDelimiterResult(FindTokenStatus.FAILED, ...) if token is
+                not found because scan exhausted `max_iterations`
         """
```

Comment:
the above should be:

        @return:
            SearchState(FindTokenStatus.FOUND, ...) if token is found
            SearchState(FindTokenStatus.REACHED_EOF, ...) if token is
                not found because the scan reached the EOF
            SearchState(FindTokenStatus.FAILED, ...) if token is
                not found because scan exhausted `max_iterations`

Reply:
Although maybe it is better to just say:

@return: SearchState object

Since it is easy for the docstring to get out of sync with the code.

Reply:
done